### PR TITLE
Test/add ios accessiblitytest

### PR DESCRIPTION
**Pull request by al1baba**
- Add ios Accessibility audit test
- Make project changes required to allow accessibility audit to run

Results from console:
image

### CODE REVIEW

**Review by bcarterterp — COMMENTED** [review]
Awesome! You mind adding a section about accessibility in this PR?

File: ios/RunnerUITests/RunnerUITests.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+final class RunnerUITests: XCTestCase {
+    func testAccessibilityAudits() throws {
+        let app = XCUIApplication()
+        app.launch()
+
+        if #available(iOS 17.0, *) {
+            try app.performAccessibilityAudit()
+        } else {
+            // Fallback on earlier versions
```
Comment (amandaroa) [inline ios/RunnerUITests/RunnerUITests.swift:18]: Consider making this comment more specific to the intention for this app. Will we not do anything with accessibility prior to iOS 17 or do some manual audit elsewhere? If so, where should they look.

File: ios/RunnerUITests/RunnerUITests.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import XCTest
+
+final class RunnerUITests: XCTestCase {
+    func testAccessibilityAudits() throws {
+        let app = XCUIApplication()
+        app.launch()
+
+        if #available(iOS 17.0, *) {
+            try app.performAccessibilityAudit()
```
Comment (amandaroa) [inline ios/RunnerUITests/RunnerUITests.swift:16]: Just out of curiosity how do you see the results of this audit? Is it in the console? Coudl you include a screenshot of the results in the description of the PR?

Comment (al1baba) [reply ios/RunnerUITests/RunnerUITests.swift:16]: Can do!

File: accessibilityscanner.png
Comment (cxw320) [inline accessibilityscanner.png:1]: Is this like a visual "key" for how the report will look from the android app? If so, is it normal to be shipping this png with the source code? Its also in the BIN folder so maybe there's no concern for files in there, I'm not sure.

**Review by cxw320 — COMMENTED** [review]
This is awesome! This is the first project I've been on where there's an automated check like this in place - I feel like we're ahead of a lot of other codebases in terms of a formal accessibility check...

Can I ask - if there are 1 or 2 items that are flagged as inaccessible, will the entire iOS accessibility unit test fail? I wonder if in the future we could set some kind of "target % of accessibility coverage" - instead of a "100% is required" to pass... In case we run into something the client wants that is non-negotiable or something. This is just a curiosity.

File: accessibilityscanner.png
Comment (al1baba) [reply accessibilityscanner.png:1]: Oh sorry, this output is from the ios app. This is specifically apple's accessiblity audit. I'm also not sure if it's a concern? I do now notice that it specifically calls out the clientapp so maybe this was a bad idea... @amandaroa

**Comment by al1baba** [comment]
> This is awesome! This is the first project I've been on where there's an automated check like this in place - I feel like we're ahead of a lot of other codebases in terms of a formal accessibility check...
> 
> Can I ask - if there are 1 or 2 items that are flagged as inaccessible, will the entire iOS accessibility unit test fail? I wonder if in the future we could set some kind of "target % of accessibility coverage" - instead of a "100% is required" to pass... In case we run into something the client wants that is non-negotiable or something. This is just a curiosity.

So I believe this whole test will fail if any issues are found. It is possible to call only certain portions of the accessibility audit to test certain things, so if we wanted, we could create separate tests for text size, contrast, button sizing etc! That would give us better reporting I think especially as the app grows...excellent callout!

File: accessibilityscanner.png
Comment (amandaroa) [reply accessibilityscanner.png:1]: I think it's fine to leave it in here for now and we need to just make a mental note (or even better yet a task in Trello)  to update it soon. Regardless that reference can be found in the git history so it's not like we are introducing anything that wasn't there before

**Review by amandaroa — APPROVED** [review]
Thanks for adding the additional comments and documentation. Looks good!

**Review by bcarterterp — APPROVED** [review]

**Pull request merged**